internal/cose.go

```diff
@@ -0,0 +1,165 @@
+package internal
+
+import (
+	"crypto/ecdsa"
+	"crypto/sha512"
+	"fmt"
+	"math/big"
+
+	"github.com/fxamacker/cbor/v2"
+)
+
+type SigningAlgorithm int
+
+const (
+	BadOrMissingAlgorithm SigningAlgorithm = iota
+	ES384
+)
+
+// https://datatracker.ietf.org/doc/html/rfc8152#section-4.2
+type CoseSign1 struct {
+	_ struct{} `cbor:",toarray"`
+
+	Protected   []byte
+	Unprotected cbor.RawMessage
+	Payload     []byte
+	Signature   []byte
+}
```

Comment:
Previously, this was referred to as `CosePayload` but it is in fact a `CoseSign1` structure as defined in the RFC I linked---this is format in which Nitro returns signed attestations.